content/part2.md

```diff
@@ -0,0 +1,186 @@
+For example, let's take a look at implementing an abandoned cart email notification.
+
+Checking for an Abandoned Cart
+------------------------------
+
+In eCommerce, an [_abandoned_ shopping cart](https://www.optimizely.com/optimization-glossary/shopping-cart-abandonment/#:~:text=Shopping%20cart%20abandonment%20is%20when,process%20before%20completing%20the%20purchase.&text=This%20rate%20will%20identify%20what,don't%20complete%20the%20purchase.) is a shopping cart that has items, but the user hasn't added
+any new items or checked out after a few hours.
+Below is an example abandoned cart email that I recently received with an offer to incentivize checkout.
+
+<img src="https://codebarbarian-images.s3.amazonaws.com/shopping-cart.jpg">
+
+With a traditional web app, abandoned cart notifications are tricky.
+You need to use a job queue like [Celery](https://en.wikipedia.org/wiki/Celery_(software\)) in Python or [Machinery](https://github.com/RichardKnop/machinery) in Golang.
+You need to schedule a job that checks if the cart is abandoned, and reschedule that job every time the cart is updated.
+
+With Temporal, you don't need a separate job queue. Instead, you define a _Selector_ with two event handlers: one that responds to a Workflow signal, and one that responds to a timer.
+By creating a new Selector on each iteration of the `for` loop, you're telling Temporal to handle the next update cart signal it receives, or send an abandoned cart email if it doesn't receive a signal for `abandonedCartTimeout`.
+Calling `Select()` on a selector blocks the Workflow until there's either a signal or `abandonedCartTimeout` elapses.
+
+```golang
+func CartWorkflow(ctx workflow.Context, state CartState) error {
+	logger := workflow.GetLogger(ctx)
+
+	err := workflow.SetQueryHandler(ctx, "getCart", func(input []byte) (CartState, error) {
+		return state, nil
+	})
+	if err != nil {
+		logger.Info("SetQueryHandler failed.", "Error", err)
+		return err
+	}
+
+	channel := workflow.GetSignalChannel(ctx, "cartMessages"
+	sentAbandonedCartEmail := false
+
+	for {
+    // Create a new selector on each iteration of the loop means Temporal will pick the first
+    // event that occurs each time: either receiving a signal, or responding to the timer.
+		selector := workflow.NewSelector(ctx)
+		selector.AddReceive(channel, func(c workflow.ReceiveChannel, _ bool) {
+			var signal interface{}
+			c.Receive(ctx, &signal)
+
+			// Handle signals for updating the cart
+		})
+
+    // If the user doesn't update the cart for `abandonedCartTimeout`, send an email
+    // reminding them about their cart. Only send the email once.
+		if !sentAbandonedCartEmail && len(state.Items) > 0 {
+			selector.AddFuture(workflow.NewTimer(ctx, abandonedCartTimeout), func(f workflow.Future) {
+				sentAbandonedCartEmail = true
+				ao := workflow.ActivityOptions{
+					ScheduleToStartTimeout: time.Minute,
```

Comment:
we are about to start heavily recommending people to use `ScheduleToCloseTimeout` over `ScheduleToStartTimeout` because this is a very niche timeout that people should not use. just making a start by correcting this

```suggestion
					ScheduleToCloseTimeout: time.Minute,
```